custom/icds_reports/filters.py

```diff
@@ -35,16 +37,22 @@ def loc_to_json(loc, project):
             'can_edit': True
         }
 
+    def user_locations():
+        user_location = user.get_sql_location(domain)
+        all_related_ancestors = user_location.get_ancestors(include_self=True)
+        return [location.location_id for location in all_related_ancestors]
+
     project = Domain.get_by_name(domain)
 
     locations = SQLLocation.root_locations(domain)
+    user_location_list = user_locations()
     if not show_test:
         locations = [
-            loc for loc in locations if loc.metadata.get('is_test_location', 'real') != 'test'
+            loc for loc in locations if loc.metadata.get('is_test_location', 'real') != 'test' and
+                                        loc.location_id in user_location_list
```

Comment:
this would skip the check for accessible locations if we are viewing test locations. I think that is not what we want. This check should be independent of test locations being viewed or not